### ISSUE TITLE
Python Auto-completion Setup

### ISSUE THREAD

**Issue by childishriver** [helix-editor/helix#2293]
Sorry, completely new to this editor, and with the lack of videos on this fairly new text editor, I require some step by step and clear help on how to setup up Python Auto-completion on Mac M1 for Helix text editor. I tried copying from the repo area, but this led to nothing. Also, am I missing a command that I should type to toggle Auto-completion?

Thank you very much, and I look forward to using this text editor and being one of the first few people to get onto this!

**Comment by the-mikedavis** [comment]
If you haven't already, you'll need to install the python language server `pylsp`: https://github.com/helix-editor/helix/wiki/How-to-install-the-default-language-servers#pylsp

With that installed, auto-complete and other language server driven features like goto-definition should work without any extra configuration or commands.

**Comment by childishriver** [comment]
> If you haven't already, you'll need to install the python language server `pylsp`: https://github.com/helix-editor/helix/wiki/How-to-install-the-default-language-servers#pylsp
> 
> With that installed, auto-complete and other language server driven features like goto-definition should work without any extra configuration or commands.

Thank you very much, that worked, is there anyway to fix error : W191, indentation contains tab

**Comment by the-mikedavis** [comment]
I'm not too familiar with python but I think it comes from this PEP https://peps.python.org/pep-0008/#tabs-or-spaces

Seems like the language server wants you to use spaces for indentation instead of tabs.

**Comment by childishriver** [comment]
> I'm not too familiar with python but I think it comes from this PEP https://peps.python.org/pep-0008/#tabs-or-spaces
> 
> Seems like the language server wants you to use spaces for indentation instead of tabs.

Is there any way I can turn off the error messages, there are so many (E303, W141, my whole code is just underlined) if so how do I do it, because the tutorial given by the website "https://github.com/python-lsp/python-lsp-server" is not very clear, and I was unable to find a file called pycodestyle in my configs.

**Comment by the-mikedavis** [comment]
Oh hmm, I'm not sure how configurable pylsp is for turning those of or tuning them. As far as I know, there isn't an option within Helix to disable diagnostics.

**Comment by lukepighetti** [comment]
In the world of python it's up to you to select the correct LSP for your needs, `helix` merely displays the results from your LSP

**Comment by childishriver** [comment]
> 

Put the exact same thing inside my languages.toml and I still got the error:
`thread 'main' panicked at 'Failed to parse WorkspaceConfiguration params: Error { code: InvalidParams, message: "Invalid params: invalid value: string \"\", expected relative URL without a base.", data: None }', helix-lsp/src/lib.rs:216:22
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace`
Is there something wrong I am doing here?

**Comment by childishriver** [comment]
> In the world of python it's up to you to select the correct LSP for your needs, `helix` merely displays the results from your LSP.
> 
> This is not true in all languages. Dart, for example, works with Helix without any configuration. Install dart, install helix, it's working. The zero-conf nature in this case is due to Dart coming prepackaged with a configured LSP.

Yes, I would like to learn more as I am yet just a beginner, a newborn coming out of the computers womb, searching merely for just answers that cannot be obtained oh so easily

**Comment by paul-scott** [comment]
@The-Rivers see #1898 for a possible work-around for your crash.
